### PR TITLE
DS-529: Pega.com: Add 'Download slides' CTA to PW Replays "Best Of" cards

### PR DESCRIPTION
**Pull request by MarcinMr**
## Jira

https://pegadigitalit.atlassian.net/browse/DS-529

## Summary

Download slides CTA was added. All phases from the `Best-of-content` were removed and only one final version was kept.

## Details

Download slides CTA was added to the `bolt-teaser` component and to the `best-of-content` pages in relation to the proposed wireframe.

## How to test

Pull the branch. Check out the `bolt-teaser` component and `best-of-content` pages where the new `Download slides` CTA was added. Make sure the new CTA is correct in terms of semantics and accessibility.

## Release notes

- Added new CTA to the Listing Teaser actions list on the `bolt-teaser` component and the `best-of-content` page.

#### Teaser component

- Action list is now aligned to the left.

### CODE REVIEW

**Review by colbytcook — APPROVED** [review]

File: docs-site/src/pages/pattern-lab/_patterns/40-components/teaser/30-teaser-status-and-actions.twig [hunk 1]
```diff
@@ -95,6 +96,29 @@
       boundary: '.js-bolt-target-teaser',
     } only %}
   {% endset %}
+  {% set download_cta %}
+    {% set icon_download %}
+      {% include '@bolt-elements-icon/icon.twig' with {
+        name: 'download',
+      } only %}
+    {% endset %}
+    {% set sr_only %}
```
Comment (mikemai2awesome) [inline docs-site/src/pages/pattern-lab/_patterns/40-components/teaser/30-teaser-status-and-actions.twig:105]: This is not needed. The visible text is already descriptive enough. Also, I do not see in the ticket's Definition of Done that says include `(PDF, 3 pages, 2.3mb)`. Is the ticket out of date?

File: docs-site/src/pages/pattern-lab/_patterns/40-components/teaser/30-teaser-status-and-actions.twig [hunk 2]
```diff
@@ -174,11 +199,35 @@
     boundary: '.js-bolt-target-teaser', // JS target for containing the popover within the teaser.
   } only %}
 {% endset %}
+{% set download_cta %}
+  {% set icon_download %}
+    {% include '@bolt-elements-icon/icon.twig' with {
+      name: 'download',
+    } only %}
+  {% endset %}
+  {% set sr_only %}
```
Comment (mikemai2awesome) [inline docs-site/src/pages/pattern-lab/_patterns/40-components/teaser/30-teaser-status-and-actions.twig:208]: Same comment as last. Not needed.

File: packages/components/bolt-teaser/teaser.schema.js [hunk 3]
```diff
@@ -104,6 +104,10 @@ module.exports = {
       type: 'object',
       description: 'Render the share button. Share menu is expected here.',
     },
+    download_cta: {
```
Comment (mikemai2awesome) [inline packages/components/bolt-teaser/teaser.schema.js:107]: The other two props don't have `_cta` postfix, so do not include it for this one. Keep in consistent and just name it `download`.

File: packages/components/bolt-teaser/src/teaser.twig [hunk 4]
```diff
@@ -105,7 +105,7 @@
             {{ status.views }} {{ 'views'|t }}
           </div>
         {% endif %}
-        {% if like or share %}
+        {% if like or share or download_cta %}
```
Comment (mikemai2awesome) [inline packages/components/bolt-teaser/src/teaser.twig:108]: Update this after the prop is renamed to `download`.

**Review by mikemai2awesome — CHANGES_REQUESTED** [review]
@MarcinMr @colbytcook I think the Best-of demos under Pages should be updated as well. Additionally, remove all the phases and keep just one final version.

File: docs-site/src/pages/pattern-lab/_patterns/40-components/teaser/30-teaser-status-and-actions.twig (hunk 1 above)
Comment (MarcinMr) [reply docs-site/src/pages/pattern-lab/_patterns/40-components/teaser/30-teaser-status-and-actions.twig]: I've read that it's a good practice to add a file name, number of pages, and weight of o file for screen readers to announce details of a downloading resource. But if it is not necessary I've already deleted those labels.

File: docs-site/src/pages/pattern-lab/_patterns/40-components/teaser/30-teaser-status-and-actions.twig (hunk 2 above)
Comment (MarcinMr) [reply docs-site/src/pages/pattern-lab/_patterns/40-components/teaser/30-teaser-status-and-actions.twig]: Deleted

File: packages/components/bolt-teaser/teaser.schema.js (hunk 3 above)
Comment (MarcinMr) [reply packages/components/bolt-teaser/teaser.schema.js]: Renamed

File: packages/components/bolt-teaser/src/teaser.twig (hunk 4 above)
Comment (MarcinMr) [reply packages/components/bolt-teaser/src/teaser.twig]: Updated

**Comment by MarcinMr** [comment]
> @MarcinMr we need to make like, share, download row left aligned. See the "New, clickable 'Download slides' CTA added to PW Replays cards added to the right of existing "Like" and "Share" CTAs (now aligned left)" definition of done.

@colbytcook changed to aligned left

**Review by mikemai2awesome — APPROVED** [review]
Excellent job! @MarcinMr

File: docs-site/src/pages/pattern-lab/_patterns/40-components/teaser/30-teaser-status-and-actions.twig (hunk 1 above)
Comment (mikemai2awesome) [reply docs-site/src/pages/pattern-lab/_patterns/40-components/teaser/30-teaser-status-and-actions.twig]: If the data exists, they can create a tooltip with the info. I'll leave that for them to decide.

**Review by danielamorse — CHANGES_REQUESTED** [review]
Looks good from a code standpoint. Nice cleanup of our old demos.

@mikemai2awesome I noticed the Listing Teaser actions can get pretty cramped, and the ones using the Icon Component vs Element wrap differently. Maybe we at least want to convert "Views" to Element for consistent wrapping?

image

@MarcinMr since this includes a visual change to the alignment of the Listing Teaser actions, would you please add that to the PR Release Notes? See [this PR](https://github.com/boltdesignsystem/bolt/pull/2310) for reference on formatting.

File: docs-site/src/pages/pattern-lab/_patterns/40-components/teaser/30-teaser-status-and-actions.twig (hunk 1 above)
Comment (mikemai2awesome) [reply docs-site/src/pages/pattern-lab/_patterns/40-components/teaser/30-teaser-status-and-actions.twig]: I asked Ross in the ticket. Will wait for his reply. https://pegadigitalit.atlassian.net/browse/WWW-858?focusedCommentId=37360

**Review by mikemai2awesome — CHANGES_REQUESTED** [review]
@danielamorse Good catch. We should fix that.

**Review by mikemai2awesome — APPROVED** [review]
I pushed a fix for it. The icon alignment was off. @colbytcook

**Review by colbytcook — APPROVED** [review]

**Review by danielamorse — APPROVED** [review]
Alignment is fixed, code looks good 👍 

1. Updated the Teaser Jest snapshots, waiting for those to pass on Travis.
2. Updated the Teaser Status and Actions demo to use the [Twig](https://github.com/boltdesignsystem/bolt/pull/2312/files#diff-f73705727454aa76dd3ab6a16ec91b54d1b38dd06fd9ef21d0665c0e0c7db4c7R105-R114) version of Tooltip instead of the [web component](https://github.com/boltdesignsystem/bolt/pull/2312/commits/bdbeaeae98ecc78b48c41b7401df124e0faf891a#diff-f73705727454aa76dd3ab6a16ec91b54d1b38dd06fd9ef21d0665c0e0c7db4c7R105-R115). We use `<bolt-tooltip>` internally in some templates because it's convenient, but for demo code that devs will reference I think we want Twig.

@colbytcook @mikemai2awesome let me know if I'm missing something regarding the second note. If not, and you approve, ship this thing! :shipit:

**Comment by MarcinMr** [comment]
> @MarcinMr since this includes a visual change to the alignment of the Listing Teaser actions, would you please add that to the PR Release Notes? See [this PR](https://github.com/boltdesignsystem/bolt/pull/2310) for reference on formatting.

@danielamorse Updated PR with release notes and visual changes

**Comment by MarcinMr** [comment]
> @mikemai2awesome I noticed the Listing Teaser actions can get pretty cramped, and the ones using the Icon Component vs Element wrap differently. Maybe we at least want to convert "Views" to Element for consistent wrapping?
> 
@danielamorse @mikemai2awesome 

There is also a small issue with the horizontal teaser on a small screen. Newly added CTA overflows the viewport.

overflow

Perhaps we should wrap the list or entire block on the really small screens like on the shots below:

list-wrapwhole-block-wrap

**Pull request merged**